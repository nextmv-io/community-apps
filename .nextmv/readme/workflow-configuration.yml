apps:
  - name: go-highs-knapsack
    scripts:
      - name: 0.sh
        silent: true
      - name: 1.sh
        skip: true
  - name: go-highs-orderfulfillment
    scripts:
      - name: 0.sh
        silent: true
      - name: 1.sh
        skip: true
  - name: go-highs-shiftscheduling
    scripts:
      - name: 0.sh
        silent: true
      - name: 1.sh
        skip: true
  - name: go-nextroute
    scripts:
      - name: 0.sh
        silent: true
      - name: 1.sh
        skip: true
  - name: java-hello-world
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: java-ortools-knapsack
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: java-ortools-routing
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-ampl-facilitylocation
    scripts:
      - name: 0.sh
        silent: true
      - name: 1.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-ampl-knapsack
    scripts:
      - name: 0.sh
        silent: true
      - name: 1.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-ampl-priceoptimization
    scripts:
      - name: 0.sh
        silent: true
      - name: 1.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-gurobi-knapsack
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-hello-world
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-hexaly-knapsack
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
      - name: 1.sh
        skip: true
  - name: python-highs-knapsack
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-nextroute
    scripts:
      - name: 0.sh
        silent: true
  - name: python-ortools-costflow
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-ortools-demandforecasting
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-ortools-knapsack
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-ortools-knapsack-multicsv
    scripts:
      - name: 0.sh
        silent: true
      - name: 1.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-ortools-routing
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-ortools-shiftassignment
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-ortools-shiftplanning
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-pyomo-knapsack
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-pyomo-shiftassignment
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-pyomo-shiftplanning
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-pyoptinterface-knapsack
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-pyvroom-routing
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
  - name: python-xpress-knapsack
    scripts:
      - name: 0.sh
        silent: true
      - name: 2.sh
        skip: true
