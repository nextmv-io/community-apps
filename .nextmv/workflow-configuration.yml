apps:
  - name: go-hello-world
    type: go
    app_id:
    marketplace_app_id:
    marketplace_major_version:
    description: Get started with Go and Nextmv.
  - name: go-highs-knapsack
    type: go
    app_id: knapsack-gosdk
    marketplace_app_id: knapsack.gosdk
    marketplace_major_version: v1
    description: Use Go and HiGHS - knapsack.
  - name: go-highs-orderfulfillment
    type: go
    app_id: order-fulfill-gosdk
    marketplace_app_id: order.fulfill.gosdk
    marketplace_major_version: v1
    description: Use Go and HiGHS - order fulfillment.
  - name: go-highs-shiftscheduling
    type: go
    app_id: shift-scheduling
    marketplace_app_id: shift.scheduling
    marketplace_major_version: v1
    description: Use Go and HiGHS - shift scheduling.
  - name: go-nextroute
    type: go
    app_id: nextroute
    marketplace_app_id: nextroute
    marketplace_major_version: v1
    description: Use Go and Nextroute - VRP.
  - name: java-hello-world
    type: java
    app_id:
    marketplace_app_id:
    marketplace_major_version:
    description: Get started with Java and Nextmv.
  - name: java-ortools-knapsack
    type: java
    app_id: knapsack-jortools
    marketplace_app_id: knapsack.jortools
    marketplace_major_version: v1
    description: Use Java and OR-Tools - knapsack.
  - name: java-ortools-routing
    type: java
    app_id:
    marketplace_app_id:
    marketplace_major_version:
    description: Use Java and OR-Tools - VRP.
  - name: python-ampl-facilitylocation
    type: python
    app_id:
    marketplace_app_id:
    marketplace_major_version:
    description: Use Python and AMPL - facility location.
  - name: python-ampl-knapsack
    type: python
    app_id: knapsack-ampl
    marketplace_app_id: knapsack.ampl
    marketplace_major_version: v1
    description: Use Python and AMPL - knapsack.
  - name: python-ampl-priceoptimization
    type: python
    app_id:
    marketplace_app_id:
    marketplace_major_version:
    description: Use Python and AMPL - price optimization.
  - name: python-gurobi-knapsack
    type: python
    app_id: knapsack-gurobi
    marketplace_app_id: knapsack.gurobi
    marketplace_major_version: v1
    description: Use Python and Gurobi - knapsack.
  - name: python-hello-world
    type: python
    app_id:
    marketplace_app_id:
    marketplace_major_version:
    description: Get started with Python and Nextmv.
  - name: python-hexaly-knapsack
    type: python
    app_id:
    marketplace_app_id:
    marketplace_major_version:
    description: Use Python and Hexaly - knapsack.
  - name: python-highs-knapsack
    type: python
    app_id: knapsack-highs
    marketplace_app_id: knapsack.highs
    marketplace_major_version: v1
    description: Use Python and HiGHS - knapsack.
  - name: python-nextroute
    type: python
    app_id:
    marketplace_app_id:
    marketplace_major_version:
    description: Use Python and Nextroute - VRP.
  - name: python-ortools-costflow
    type: python
    app_id: cost-flow-ortools
    marketplace_app_id: cost.flow.ortools
    marketplace_major_version: v1
    description: Use Python and OR-Tools - cost flow.
  - name: python-ortools-demandforecasting
    type: python
    app_id: demand-forecasting-ortools
    marketplace_app_id: dem.forecast.ortools
    marketplace_major_version: v1
    description: Use Python and OR-Tools - demand forecasting.
  - name: python-ortools-knapsack
    type: python
    app_id: knapsack-ortools
    marketplace_app_id: knapsack.ortools
    marketplace_major_version: v1
    description: Use Python and OR-Tools - knapsack.
  - name: python-ortools-knapsack-multicsv
    type: python
    app_id:
    marketplace_app_id:
    marketplace_major_version:
    description: Use Python and OR-Tools - multi CSV knapsack.
  - name: python-ortools-routing
    type: python
    app_id: routing-ortools
    marketplace_app_id: routing.ortools
    marketplace_major_version: v1
    description: Use Python and OR-Tools - VRP.
  - name: python-ortools-shiftassignment
    type: python
    app_id: shift-assign-ortools
    marketplace_app_id: shift.assign.ortools
    marketplace_major_version: v1
    description: Use Python and OR-Tools - shift assignment.
  - name: python-ortools-shiftplanning
    type: python
    app_id: shift-plan-ortools
    marketplace_app_id: shift.plan.ortools
    marketplace_major_version: v1
    description: Use Python and OR-Tools - shift planning.
  - name: python-pyomo-knapsack
    type: python
    app_id: knapsack-pyomo
    marketplace_app_id: knapsack.pyomo
    marketplace_major_version: v1
    description: Use Python and Pyomo - knapsack.
  - name: python-pyomo-shiftassignment
    type: python
    app_id: shift-assign-pyomo
    marketplace_app_id: shift.assign.pyomo
    marketplace_major_version: v1
    description: Use Python and Pyomo - shift assignment.
  - name: python-pyomo-shiftplanning
    type: python
    app_id: shift-plan-pyomo
    marketplace_app_id: shift.plan.pyomo
    marketplace_major_version: v1
    description: Use Python and Pyomo - shift planning.
  - name: python-pyoptinterface-knapsack
    type: python
    app_id:
    marketplace_app_id:
    marketplace_major_version:
    description: Use Python and PyOptInterface - knapsack.
  - name: python-pyvroom-routing
    type: python
    app_id: routing-pyvroom
    marketplace_app_id: routing.pyvroom
    marketplace_major_version: v1
    description: Use Python and Pyvroom - VRP.
  - name: python-xpress-knapsack
    type: python
    app_id:
    marketplace_app_id:
    marketplace_major_version:
    description: Use Python and Xpress - knapsack.

# Main supported language versions.
language-support:
  go:
    version: 1.22
  python:
    version: 3.11
  java:
    version: 21
    distribution: temurin
