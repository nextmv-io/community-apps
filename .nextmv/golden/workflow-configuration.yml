apps:
  - name: go-highs-knapsack
    language: go # we always use our main version
  - name: go-highs-orderfulfillment
    language: go # we always use our main version
  - name: go-highs-shiftscheduling
    language: go # we always use our main version
  - name: go-nextroute
    language: go # we always use our main version
  - name: java-ortools-knapsack
    language: java
    language-version: 21
    distribution: temurin
  - name: python-ampl-facilitylocation
    language: python
    language-version: 3.11
  - name: python-ampl-knapsack
    language: python
    language-version: 3.11
  - name: python-ampl-priceoptimization
    language: python
    language-version: 3.11
  - name: python-gurobi-knapsack
    language: python
    language-version: 3.11
  - name: python-hello-world
    language: python
    language-version: 3.11
  - name: python-highs-knapsack
    language: python
    language-version: 3.11
  - name: python-nextroute
    language: python
    language-version: 3.11
  - name: python-ortools-costflow
    language: python
    language-version: 3.11
  - name: python-ortools-knapsack
    language: python
    language-version: 3.11
  - name: python-ortools-routing
    language: python
    language-version: 3.11
  - name: python-ortools-shiftassignment
    language: python
    language-version: 3.11
  - name: python-ortools-shiftplanning
    language: python
    language-version: 3.11
  - name: python-pyomo-knapsack
    language: python
    language-version: 3.11
  - name: python-pyomo-shiftassignment
    language: python
    language-version: 3.11
  - name: python-pyomo-shiftplanning
    language: python
    language-version: 3.11
  - name: python-pyoptinterface-knapsack
    language: python
    language-version: 3.11
  - name: python-pyvroom-routing
    language: python
    language-version: 3.11
  - name: python-xpress-knapsack
    language: python
    language-version: 3.11
